### PR TITLE
WEB-1492 Checkbox "column" is moveable in backdraft Manage pages

### PR DESCRIPTION
**Pull request by codingiam**

### CODE REVIEW

File: src/plugins/data_table/column_config_generator.js [hunk 1]
```diff
@@ -31,6 +31,11 @@ var ColumnConfigGenerator =  Backdraft.Utils.Class.extend({
       }
     });
 
+    // make the bulk column the first one if present
+    this.columnsConfig = _.sortBy(this.columnsConfig, function (columnConfig) {
```
Comment (nburwell) [inline src/plugins/data_table/column_config_generator.js]: looks like we need to tweak HoundCI to know about some global variables. @HParker can you point me to where we can do that?

File: dist/backdraft.js [hunk 2]
```diff
@@ -527,6 +527,11 @@ _.extend(Plugin.factory, {
       }
     });
 
+    // make the bulk column the first one if present
+    this.columnsConfig = _.sortBy(this.columnsConfig, function (columnConfig) {
+      return !columnConfig.bulk;
+    });
```
Comment (codingiam) [inline dist/backdraft.js]: @nburwell With this the bulk column gets moved in the first position.

File: dist/backdraft.js [hunk 3]
```diff
@@ -1252,7 +1257,10 @@ _.extend(Plugin.factory, {
           self._columnManager.columnsSwapped(fromIndex, toIndex);
           // pass event up
           self._onColumnReorder();
-        }
+        },
+        // iFixedColumns configures how many columns should be unmovable starting from left
+        // if the first column is the bulk column we make it unmovable
+        iFixedColumns: this.$el.find("th:first.bulk :checkbox").length
```
Comment (codingiam) [inline dist/backdraft.js]: @nburwell We use this selector in other places so I think it should be safe to use for this situation.

File: dist/backdraft.js (hunk 2 above)
Comment (nburwell) [inline dist/backdraft.js]: ah, cool good to know

File: dist/backdraft.js (hunk 3 above)
Comment (nburwell) [inline dist/backdraft.js]: fair enough.. what about making the `th.first.bulk` a "constant", ie `this.BULK_COLUMN_SELECTOR`  and using throughout?

Comment (codingiam) [inline dist/backdraft.js]: I'll do that.

File: src/plugins/data_table/column_config_generator.js (hunk 1 above)
Comment (HParker) [inline src/plugins/data_table/column_config_generator.js]: https://houndci.com/configuration
https://github.com/Invoca/backdraft/blob/master/.hound.yml

Comment (nburwell) [inline src/plugins/data_table/column_config_generator.js]: Thanks, added to master. Going to merge in here and see if we get any hounding

**Comment by codingiam** [comment]
This is ready for rereview.

**Comment by nburwell** [comment]
:+1:

**Pull request merged**